### PR TITLE
Ana/826 add popular selector for validators

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes luniehq/lunie-api#826

**Description:**

Just opening to let you know I'm now working on this

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by Bitcoinera** [comment]
Working and looking like this:

image

Probably not the best design though. I just put a star and didn't use a css variable for the star color. I don't know very well yet how to create cool variables that suit all themese, but will give it a try.

Also, another mystery: why in production we are not getting any request for ValidatorPage views 🤷

**Comment by Bitcoinera** [comment]
Could also move the getValidatorViews logic to the `block-listener`, if we need to use this `popularity` field somewhere else in the API 🤔

**Comment by faboweb** [comment]
The issue states: "Add default "popular" selector for validators in Lunie"
Please remove the "popular" column and add a selector left of "All" which has the caption "Popular". This will filter all validators out which have no image. The list should always be sorted by default by the validator views (which we store in the db).

**Comment by Bitcoinera** [comment]
> The issue states: "Add default "popular" selector for validators in Lunie"
> Please remove the "popular" column and add a selector left of "All" which has the caption "Popular". This will filter all validators out which have no image. The list should always be sorted by default by the validator views (which we store in the db).

Oh, ok! My reading comprehension 🤦

**Comment by faboweb** [comment]
🚀 Deployed on https://5ede7f3023c74408162990a7--lunieio.netlify.app

**Comment by Bitcoinera** [comment]
> 🚀 Deployed on https://5ede7f3023c74408162990a7--lunieio.netlify.app

Nice! So now the Netlify builds are per commit. So cool

**Comment by Bitcoinera** [comment]
OK, this one is ready to go! Working

Just one question, if popular is `false`, we don't invert the list right? Then I just return the normal list

**Comment by Bitcoinera** [comment]
Ah, forgot the images thing. Rolling it out

**Comment by faboweb** [comment]
Please do all filtering and sorting server side.

**Comment by faboweb** [comment]
If popular is false we still show sorted by popularity but we show all

**Comment by Bitcoinera** [comment]
> If popular is false we still show sorted by popularity but we show all

Oh, ok. So we always sort by popularity then 👍 Roger that

**Comment by Bitcoinera** [comment]
all working. It just takes a little while to load the first time the request is made. But then everything runs smoothly

**Comment by Bitcoinera** [comment]
Let's see if now E2E pass. Should we kill codecov?

This one is ready to go! 💃

**Comment by faboweb** [comment]
API changes need to be put in another PR so we can merge and release it individually :/

**Comment by faboweb** [comment]
tested and working

**Comment by Bitcoinera** [comment]
> tested and working

Of course! What did you expect :stuck_out_tongue: ?

File: app/src/components/staking/PageValidators.vue [hunk 1]
```diff
@@ -88,20 +88,18 @@ export default {
...
-        this.popularSort = false
-        this.activeOnly = false
+        this.allValidators = true
       }
       if (selector === `activeOnly`) {
-        this.activeOnly = !this.activeOnly
-        this.popularSort = false
-        this.allValidators = false
+        this.activeOnly = true
       }
```
Comment (Bitcoinera) [inline app/src/components/staking/PageValidators.vue:103]: Ah yeah, this looks much better like this. Thank you!

**Comment by Bitcoinera** [comment]
Yeah, this never passed the E2E actually, so there you go :thinking:

**Pull request merged**